### ISSUE TITLE
Add option for `finalize` to not swallow error's for testing purposes.

### ISSUE THREAD

**Issue by jakebolewski** [JuliaLang/julia#13775]
Currently when there is an error thrown in a finalizer, the error is swallowed and an error message is printed.

```julia
julia> type Foo
       end

julia> f = Foo()
Foo()

julia> finalizer(f, x -> error("foo"))

julia> finalize(f)
error in running finalizer: ErrorException("foo")
```

Could we add an option for the error not to get swallowed for testing purposes?   When bugs are introduced into the finalizer function most CI tests will silently pass.

**Comment by JeffBezanson** [comment]
Would the exception get thrown from wherever GC happened to occur, or only the `finalize` function?

**Comment by jakebolewski** [comment]
The non-locality of throwing the error whenever GC happened seems a bit problematic, I was thinking of just adding an extra parameter to `finalize(x, raise::Bool=false)` which you could then just call explicitly in the test suite with `finalize(x, true)`.

**Comment by JeffBezanson** [comment]
Agreed; it seems totally fine for this to just be part of `finalize`.
